tools/autograd/load_derivatives.py

```diff
@@ -144,15 +144,25 @@ def set_up_derivatives(defn_name, defn, declaration):
 
         # Set up the derivative information
         derivatives = []
+        not_derivatives = []
+        not_args_with_gradients = []
         for raw_names in sorted(defn.keys()):
             formula = defn[raw_names]
             names = split_names(raw_names)
-            derivatives.append(create_derivative(declaration, formula, names))
+            if formula.lower().strip() == 'not_differentiable':
+                derivative = create_derivative(declaration['arguments'], declaration['returns'],
```

Comment:
since you unconditionally create the derivative can you not put it in an if / else.